lib/vrl/stdlib/src/chunks.rs

```diff
@@ -0,0 +1,124 @@
+use ::value::Value;
+use vrl::prelude::*;
+
+fn chunks(value: Value, chunk_size: Value) -> Resolved {
+    let chunk_size = chunk_size.try_integer()?;
+    let bytes = value.try_bytes()?;
+
+    if chunk_size < 1 {
+        return Err(r#""chunk_size" must be at least 1 byte"#.into());
+    }
+
+    Ok(bytes.chunks(chunk_size as usize).collect::<Vec<_>>().into())
```

Comment:
In previous commits, I was casting `chunk_size` to `usize` in the `let` expression, but it led to the discovery that negative numbers were being cast to `usize::MAX` (see this issue: https://github.com/vectordotdev/vector/issues/13891). Casting it to `usize` here means allows the chunk size check to occur on [L9](https://github.com/vectordotdev/vector/pull/13794/files#diff-2166716f04399fa4177d788683336e029a5e889207341eca092761ff5a585e1dR9) before being cast to `usize`.

However, it does raise a question: will vector always be compiled on a 64-bit target architecture? e.g. 32-bit architectures would have a smaller implicit maximum `chunk_size`.

Reply:
Thanks for checking this!

Vector can be compiled for 32-bit targets. You can use `usize::try_from` to see if a value can be correctly converted. It should also be added to the type definition, so valid literals won't make the function fallible. You can also do a check with `usize::BITS` and if it's >=64, the check isn't necessary even if the `chunk_size` isn't a literal.